### PR TITLE
Add xml2json and json2xml functions

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Adds two new functions
xml2json and json2xml that will help in conversions between xml and json data strings
Uses this library: https://github.com/nashwaan/xml-js

Implements #2015 

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
- Cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/constants/defs/xml2json.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "!name": "xml-js",
+  "xml2json": {
+    "!doc": "converts xml string to json string",
```
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/defs/xml2json.json:4]: please add this to docs :)

Comment (Nikhil-Nandagopal) [reply app/client/src/constants/defs/xml2json.json:4]: especially the function signature

**Comment by hetunandu** [comment]
Docs PR opened here appsmithorg/appsmith-docs#69

File: app/client/src/constants/defs/xml2json.json (hunk 1 above)
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/defs/xml2json.json:4]: @hetunandu this should convert it to a js object not a string

File: app/client/src/constants/defs/xml2json.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "!type": "fn(xml: string, options: object) -> string"
```
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/defs/xml2json.json:5]: options should be an optional arg and it should function without it right?

File: app/client/src/constants/defs/xml2json.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  },
+  "json2xml": {
+    "!doc": "converts json string to xml string",
+    "!type": "fn(json: string, options: object) -> string"
```
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/defs/xml2json.json:9]: same here options should be optional

File: app/client/src/constants/defs/xml2json.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "!name": "xml-js",
+  "xml2json": {
+    "!doc": "converts xml string to json object",
+    "!type": "fn(xml: string, options?: object) -> object"
```
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/defs/xml2json.json:5]: @hetunandu I tried this out and I dislike the output of this library. It feels very complicated and not usable.
Can we use this instead
https://github.com/NaturalIntelligence/fast-xml-parser
This is a cleaner output

**Review by Nikhil-Nandagopal — CHANGES_REQUESTED** [review]
Left a comment on the lib we're using

**Review by Nikhil-Nandagopal — APPROVED** [review]

**Pull request merged**